pkg/cpioparser/cpioparser.go

```diff
@@ -0,0 +1,238 @@
+	var fi fsparser.FileInfo
+	size, _ := strconv.Atoi(res[0][5])
+	if nameIdx == 7 {
+		fi.Size = int64(size)
+	}
+	fi.Mode, _ = parseMode(res[0][1])
+	fi.Uid, _ = strconv.Atoi(res[0][3])
+	fi.Gid, _ = strconv.Atoi(res[0][4])
+	// cpio returns relative pathnames so add leading "/"
+	fi.Name = "/" + res[0][nameIdx]
+
+	// fill in linktarget
+	if fi.IsLink() && strings.Contains(fi.Name, "->") {
+		rlnk := p.fileLinkReg.FindAllStringSubmatch(fi.Name, -1)
+		if rlnk == nil {
+			return "", fsparser.FileInfo{}, fmt.Errorf("can't parse LinkTarget from %s", fi.Name)
+		}
+		fi.Name = rlnk[0][1]
+		fi.LinkTarget = rlnk[0][2]
+	}
+
+	// handle root directory
+	if fi.Name == "/." {
+		dirpath = "."
+		fi.Name = "."
+	} else {
+		dirpath, fi.Name = normalizePath(fi.Name)
+	}
+
+	return dirpath, fi, nil
+
+}
+
+// GetDirInfo returns information on the specified directory.
+func (p *CpioParser) GetDirInfo(dirpath string) ([]fsparser.FileInfo, error) {
+	if err := p.loadFileList(); err != nil {
+		return nil, err
+	}
+
+	return p.files[path.Clean(dirpath)], nil
+}
+
+// GetFileInfo returns information on the specified file.
+func (p *CpioParser) GetFileInfo(filepath string) (fsparser.FileInfo, error) {
+	if err := p.loadFileList(); err != nil {
+		return fsparser.FileInfo{}, err
+	}
+
+	dirpath, name := normalizePath(filepath)
+	// the root is stored as "."
+	if dirpath == "/" && name == "" {
+		dirpath = "."
+		name = "."
+	}
+	dir := p.files[dirpath]
+	for _, fi := range dir {
+		if fi.Name == name {
+			return fi, nil
+		}
+	}
+	return fsparser.FileInfo{}, fmt.Errorf("Can't find file %s", filepath)
+}
+
+func (p *CpioParser) loadFileList() error {
+	if p.files != nil {
+		return nil
+	}
+	p.files = make(map[string][]fsparser.FileInfo)
+
+	out, err := exec.Command("sh", "-c", cpioCmd+" -tvn < "+p.imagepath).CombinedOutput()
+	if err != nil {
+		if err.Error() != errors.New("exit status 2").Error() {
+			fmt.Fprintf(os.Stderr, "getDirList: >%s<", err)
+			return err
+		}
+	}
+
+	lines := strings.Split(string(out), "\n")
+	for _, line := range lines {
+		if len(line) < 11 {
+			continue
+		}
+		if strings.HasPrefix(line, "cpio") {
+			continue
+		}
+		fmt.Println(line)
+		path, fi, err := p.parseFileLine(line)
+		if err == nil {
+			dirfiles := p.files[path]
+			dirfiles = append(dirfiles, fi)
+			p.files[path] = dirfiles
+		}
+	}
+	return nil
+}
+
+// CopyFile copies the specified file to the specified destination.
+func (p *CpioParser) CopyFile(filepath string, dstdir string) bool {
+	out, err := exec.Command("sh", "-c", cpioCmd+" -i --to-stdout "+filepath[1:]+" < "+p.imagepath+" > "+dstdir).CombinedOutput()
+	if err != nil {
+		if err.Error() != errors.New("exit status 2").Error() {
+			fmt.Fprintf(os.Stderr, "cpio failed: %v: %s\n", err, out)
+			return false
+		}
+	}
+	return true
+}
+
+func (p *CpioParser) Supported() bool {
+	_, err := exec.LookPath(cpioCmd)
+	if err != nil {
+		return false
+	}
+	_, err = exec.LookPath(cpCmd)
```

Comment:
this is fine but i think it would be easier to read this function if it matched the above

```go
if _, err := ...; err != nil {
    return true
}
return false
```